### PR TITLE
feat(sqs): implement inbound connector

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## Description

 implement inbound AWS SQS  connector

Screenshot 2023-05-05 at 12 54 11
Screenshot 2023-05-05 at 12 54 18

closes #
https://github.com/camunda/connectors-bundle/issues/498

### CODE REVIEW

File: connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.api.inbound.InboundConnectorResult;
+import io.camunda.connector.inbound.model.SqsInboundProperties;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SqsQueueConsumer {
```
Comment (igpetrov) [inline connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java:20]: I cannot see unit tests for this code. Is it covered?

**Review by igpetrov — APPROVED** [review]
LGTM

File: connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsExecutable.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    amazonSQS =
+        sqsClientSupplier.sqsClient(
+            properties.getAuthentication().getAccessKey(),
+            properties.getAuthentication().getSecretKey(),
+            properties.getQueue().getRegion());
+
+    SqsQueueConsumer sqsQueueConsumer = new SqsQueueConsumer(amazonSQS, properties, context);
+
+    sqsQueueConsumer.consumeQueueUntilActivated();
```
Comment (sbuettner) [inline connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsExecutable.java:51]: Should be named `consumeQueueUntilDeactivated` I guess.

File: connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  private final InboundConnectorContext context;
+
+  public SqsQueueConsumer(
+      AmazonSQS sqsClient, SqsInboundProperties properties, InboundConnectorContext context) {
+    this.sqsClient = sqsClient;
+    this.properties = properties;
+    this.context = context;
+  }
+
+  public void consumeQueueUntilActivated() {
```
Comment (sbuettner) [inline connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java:35]: As above: Should be called `consumeQueueUntilDeactivated`.

File: connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    LOGGER.info("Started SQS consumer for queue {}", properties.getQueue().getName());
+    boolean isNotActivated = true;
```
Comment (sbuettner) [inline connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java:37]: maybe just name it `active = true`?

File: connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+          properties.getQueue().getMessageAttributeNames());
+    }
+    do {
+      ReceiveMessageResult receiveMessageResult = sqsClient.receiveMessage(receiveMessageRequest);
+
+      List<Message> messages = receiveMessageResult.getMessages();
+      for (Message message : messages) {
+        InboundConnectorResult<?> correlate = context.correlate(message);
+        if (correlate.isActivated()) {
+          isNotActivated = false;
```
Comment (sbuettner) [inline connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java:56]: Do we close this subscription here then? Shouldnt it stay open for new messages?

File: connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    if (properties.getQueue().isContainMessageAttributeNames()) {
+      receiveMessageRequest.withMessageAttributeNames(
+          properties.getQueue().getMessageAttributeNames());
+    }
+    do {
+      ReceiveMessageResult receiveMessageResult = sqsClient.receiveMessage(receiveMessageRequest);
+
+      List<Message> messages = receiveMessageResult.getMessages();
+      for (Message message : messages) {
+        InboundConnectorResult<?> correlate = context.correlate(message);
```
Comment (sbuettner) [inline connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsQueueConsumer.java:54]: I couldnt find the code that acknowledges the messages after successufl correlation. Where do we do this?

File: connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsExecutable.java (hunk 2 above)
Comment (sbuettner) [reply connectors/sqs/src/main/java/io/camunda/connector/inbound/SqsExecutable.java:51]: We will also block here right? We should start the client consumer, process messages in the background and stop the process when deactivate is called.

**Review by sbuettner — APPROVED** [review]

**Pull request merged**